src/ui/resource/Image.js

```diff
@@ -457,27 +457,19 @@ exports = Class(lib.PubSub, function () {
 			return;
 		}
 
+		var argumentCount = arguments.length;
 		var map = this._map;
 		var srcImg = this._srcImg;
-		var args1 = arguments[1];
-		var args2 = arguments[2];
-		var args3 = arguments[3];
-		var args4 = arguments[4];
-		var args5 = arguments[5];
-		var args6 = arguments[6];
-		var args7 = arguments[7];
-		var args8 = arguments[8];
-		var args9 = arguments[9];
 		var srcX = map.x;
 		var srcY = map.y;
 		var srcW = map.width;
 		var srcH = map.height;
-		var destX = args5 !== void 0 ? args5 : args1 || 0;
-		var destY = args6 !== void 0 ? args6 : args2 || 0;
-		var destW = args7 !== void 0 ? args7 : args3 || 0;
-		var destH = args8 !== void 0 ? args8 : args4 || 0;
+		var destX = argumentCount > 5 ? arguments[5] : arguments[1];
+		var destY = argumentCount > 6 ? arguments[6] : arguments[2];
+		var destW = argumentCount > 7 ? arguments[7] : arguments[3];
+		var destH = argumentCount > 8 ? arguments[8] : arguments[4] || 0;
```

Comment:
I tried to leave the logic identical to the original, however we can
revisit this if need be.
On Jul 6, 2015 12:35 PM, "Martin Hunt" <notifications@github.com> wrote:

> In src/ui/resource/Image.js
> <https://github.com/gameclosure/timestep/pull/53#discussion_r33973195>:
>
> >  		var srcX = map.x;
> >  		var srcY = map.y;
> >  		var srcW = map.width;
> >  		var srcH = map.height;
> > -		var destX = args5 !== void 0 ? args5 : args1 || 0;
> > -		var destY = args6 !== void 0 ? args6 : args2 || 0;
> > -		var destW = args7 !== void 0 ? args7 : args3 || 0;
> > -		var destH = args8 !== void 0 ? args8 : args4 || 0;
> > +		var destX = argumentCount > 5 ? arguments[5] : arguments[1];
> > +		var destY = argumentCount > 6 ? arguments[6] : arguments[2];
> > +		var destW = argumentCount > 7 ? arguments[7] : arguments[3];
> > +		var destH = argumentCount > 8 ? arguments[8] : arguments[4] || 0;
>
> seems like we want the || 0 on all of the preceding 4 lines? (or none of
> them)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gameclosure/timestep/pull/53/files#r33973195>.
>